src/System.Net.Requests/src/System/Net/FtpWebRequest.cs

```diff
@@ -1688,11 +1679,7 @@ public override WebHeaderCollection Headers
         {
             get
             {
-                if (_ftpRequestHeaders == null)
-                {
-                    _ftpRequestHeaders = new WebHeaderCollection();
-                }
-                return _ftpRequestHeaders;
+                return LazyInitializer.EnsureInitialized(ref _ftpRequestHeaders, () => new WebHeaderCollection());
```

Comment:
This wasn't previously using synchronization, so we don't need to add any.  You can revert this.